### ISSUE TITLE
Support for approx

### ISSUE THREAD

**Issue by RestitutorOrbis** [paupino/rust-decimal#309]
Is there any interest in adding support for [approx](https://docs.rs/approx/0.4.0/approx/)? Obviously, you can implement approximate equality without drawing in another crate but it's nice to have it built-in to a library IMO. I'm willing to write up the implementation myself, I just wanted to get confirmation to see if adding a PR to address this is in line with the vision for this project.

**Comment by paupino** [comment]
It's a really good question. I've actually been thinking about this general topic a lot lately.

The library is now starting to be used for more mathematical related functions that have been typically reserved for floats.  I've been contemplating splitting out this functionality into a feature crate within rust-decimal. By doing so, it makes adding features such as this very simple and straight forward without confusing the fixed-precision foundations and making the base library too "heavy". 

Long story short - yes I would consider it, however I may see if I can break out the float-like behavior into it's own feature that can be pulled in to ensure this is a cleaner form of opt-in behavior.
